flutter-studio/src/io/flutter/profiler/FlutterStudioMonitorStageView.java

```diff
@@ -703,6 +738,61 @@ public void selectionCleared() {
     axisPanel.setOpaque(false);
     axisPanel.add(yAxisBytes, BorderLayout.WEST);
 
+    // Render the GC glyphs.
+    DurationDataModel<GcDurationData> durationModel = new DurationDataModel<>(new RangedSeries<>(viewRange, new GcStatsDataSeries(gcDataSeries)));
+    durationModel.setAttachedSeries(maxMemoryRange, Interpolatable.SegmentInterpolator);
+    durationModel.update(0);
+
+    // Build the GC icon rendering.
+    DurationDataRenderer<GcDurationData> gcRenderer = new DurationDataRenderer.Builder<>(durationModel, Color.BLACK)
+      .setIcon(StudioIcons.Profiler.Events.GARBAGE_EVENT)
+      // Need to offset the GcDurationData by the margin difference between the overlay component and the
+      // line chart. This ensures we are able to render the Gc events in the proper locations on the line.
+      .setLabelOffsets(-StudioIcons.Profiler.Events.GARBAGE_EVENT.getIconWidth() / 2f,
+                       StudioIcons.Profiler.Events.GARBAGE_EVENT.getIconHeight() / 2f)
+      .setHostInsets(new Insets(Y_AXIS_TOP_MARGIN, 0, 100, 0))
+      // TODO(terry): Need to display number of bytes reclaimed e.g., QC(.07 MB) when handleGCEvent is called (doesn't work yet).
+      .setLabelProvider(data -> String.format("GC"))
+      .build();
+
+    mLineChart.addCustomRenderer(gcRenderer);
+    overlayComponent.addDurationDataRenderer(gcRenderer);
+
+    // Build the Reset icon rendering.
+    DurationDataModel<ResetData> resetModel = new DurationDataModel<>(new RangedSeries<>(viewRange, new ResetStatsDataSeries(resetDataSeries)));
+    resetModel.setAttachedSeries(maxMemoryRange, Interpolatable.SegmentInterpolator);
+    resetModel.update(0);
+
+    DurationDataRenderer<ResetData> resetRenderer = new DurationDataRenderer.Builder<>(resetModel, Color.BLACK)
+      .setIcon(FlutterIcons.ResetMemoryStats)
+      .setHostInsets(new Insets(Y_AXIS_TOP_MARGIN, 0, 0, 0))
+      .build();
+
+    mLineChart.addCustomRenderer(resetRenderer);
+    overlayComponent.addDurationDataRenderer(resetRenderer);
+
+    // Build the Snapshot icon rendering.
+    DurationDataModel<SnapshotData> snapshotModel = new DurationDataModel<>(new RangedSeries<>(viewRange, new SnapshotDataSeries(snapshotDataSeries)));
+    snapshotModel.setAttachedSeries(maxMemoryRange, Interpolatable.SegmentInterpolator);
+    snapshotModel.update(0);
+
+    DurationDataRenderer<SnapshotData> snapshotRenderer = new DurationDataRenderer.Builder<>(snapshotModel, Color.BLACK)
+      .setIcon(FlutterIcons.Snapshot)
+      .setHostInsets(new Insets(Y_AXIS_TOP_MARGIN, 0, 0, 0))
+      .build();
+
+    mLineChart.addCustomRenderer(snapshotRenderer);
+    overlayComponent.addDurationDataRenderer(snapshotRenderer);
+
+    final JPanel overlayPanel = new JBPanel(new BorderLayout());
+    overlayPanel.setOpaque(false);
+    overlayPanel.setBorder(BorderFactory.createEmptyBorder(0, 0, 0, 0));
```

Comment:
nit: is it necessary to set an empty border?

Reply:
I don't think so - removed.